File: _scratch/run/reverse_vowels_of_a_string/sol.py
class Solution:
    def reverseVowels(self, s):
        v = set("aeiouAEIOU")
        a = list(s)
        l, r = 0, len(a) - 1
        while l < r:
            if a[l] not in v:
                l += 1
            elif a[r] not in v:
                r -= 1
            else:
                a[l], a[r] = a[r], a[l]
                l += 1
                r -= 1
        return "".join(a)
import sys
_l=sys.stdin.read().split('\n')
_T=int(_l[0]);_o=[]
for _i in range(_T):
    _s=_l[1+_i] if 1+_i<len(_l) else ""
    _o.append(str(Solution().reverseVowels(_s)))
print('\n'.join(_o))

File: _scratch/run/reverse_vowels_of_a_string/sol.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseVowels = function(s) {
    const v = "aeiouAEIOU";
    const a = s.split('');
    let l = 0, r = a.length - 1;
    while (l < r) {
        if (v.indexOf(a[l]) < 0) l++;
        else if (v.indexOf(a[r]) < 0) r--;
        else { const t = a[l]; a[l] = a[r]; a[r] = t; l++; r--; }
    }
    return a.join('');
};
const _l=require('fs').readFileSync(0,'utf8').split('\n');const _T=+_l[0];const _o=[];
for(let _i=0;_i<_T;_i++){const _s=(_l[1+_i]||'').replace(/\r$/,'');_o.push(String(reverseVowels(_s)));}
console.log(_o.join('\n'));